File: 2024/src/01/day01b.py
lefties = []
righties = []

# Read input
with open("input/day01.in", "r") as file:
    for line in file:
        parts = line.strip().split()
        l, r = map(int, parts)
        lefties.append(l)
        righties.append(r)

# Process
result = sum(righties.count(l) * l for l in lefties)
print(result)

File: 2024/src/01/day01b.js
const fs = require("fs");

lefties = [];
righties = [];
// Open the file and read it line by line
fs.readFile("input/day01.in", "utf8", (err, data) => {
  if (err) throw err;
  const lines = data.split("\n");
  lines.forEach((line) => {
    let parts = line.trim().split();
    let [l, r] = parts[0].split("  ").map(Number);
    lefties.push(l);
    righties.push(r);
  });

  // Process
  let result = 0;
  lefties.forEach((l) => {});
  lefties.forEach((l) => {
    const count = righties.filter((r) => r === l).length;
    result += count * l;
  });
  console.log(result);
});
